### PR TITLE
Add new logic for the virtual Front page

### PR DESCRIPTION
**Pull request by mitogh**
**Ticket**: https://central.tri.be/issues/100832

Make sure the `show_in_front` is set to be a `page` event as a virtual page using a negative ID to identify
this page.

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $option is never used | [Front_Page_View.php:142](https://github.com/moderntribe/the-events-calendar/blob/720c7e5fa4de476f5576d6eb023fcd5eb5999fa6/src/Tribe/Front_Page_View.php#L142)
:warning: | The method parameter $pages is never used | [Front_Page_View.php:162](https://github.com/moderntribe/the-events-calendar/blob/720c7e5fa4de476f5576d6eb023fcd5eb5999fa6/src/Tribe/Front_Page_View.php#L162)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Admin/Front_Page_View.php [hunk 1]
```diff
@@ -1,52 +0,0 @@
-<?php
-class Tribe__Events__Admin__Front_Page_View {
```
Comment (bordoni) [inline src/Tribe/Admin/Front_Page_View.php]: Move this file to the `deprecated` folder. Don't remove it just yet.

File: src/Tribe/Customizer/Front_Page_View.php [hunk 2]
```diff
@@ -1,76 +0,0 @@
...
-
-/**
- * The Events Calendar Customizer
- * Add "Main Events Page" to Homepage Selection
- *
- * @package The Events Calendar
- * @subpackage Customizer
- * @since 4.6.12
- */
-class Tribe__Events__Customizer__Front_Page_View extends Tribe__Customizer__Section {
```
Comment (bordoni) [inline src/Tribe/Customizer/Front_Page_View.php]: Move this file to the `deprecated` folder. Don't remove it just yet.

File: src/Tribe/Front_Page_View.php [hunk 3]
```diff
@@ -3,11 +3,29 @@
...
+
+	/**
+	 * The ID used to identify the virtual page, using a -10 for no particular reason, but avoding -1 as is regular
+	 * used as infinite or any other popular reference.
+	 *
+	 * @since TBD
+	 *
+	 * @var int
+	 */
+	private $HOME_VIRTUAL_ID = -10;
```
Comment (bordoni) [inline src/Tribe/Front_Page_View.php]: Lets not use uppercase variable, that is reserved for Constants. Just use a normal prop.

File: src/Tribe/Front_Page_View.php [hunk 4]
```diff
@@ -103,4 +126,130 @@ protected function get_main_events_url() {
...
+
+	/**
+	 * Make sure to set the correct values if we need to update old versions using the previous logic.
+	 *
+	 * @since TBD
+	 */
+	public function backwards_compatible() {
+
+		if ( $this->is_page_on_front() ) {
+			return;
```
Comment (bordoni) [inline src/Tribe/Front_Page_View.php]: Bail with return, so we can test this later.

File: src/Tribe/Front_Page_View.php [hunk 5]
```diff
@@ -103,4 +126,130 @@ protected function get_main_events_url() {
...
+		}
+
+		// If the archive option is false we don't need to update anything
+		if ( $this->has_event_archive_option() && 'post' === get_option( 'show_on_front' ) ) {
+			update_option( 'show_on_front', 'page' );
```
Comment (bordoni) [inline src/Tribe/Front_Page_View.php]: Use a `$let` variable to return if we updated or not. To allow this to be tested later.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $option is never used | [Front_Page_View.php:142](https://github.com/moderntribe/the-events-calendar/blob/d5f0504a24e463d6373ef772f244462a2af35dfd/src/Tribe/Front_Page_View.php#L142)
:warning: | The method parameter $pages is never used | [Front_Page_View.php:162](https://github.com/moderntribe/the-events-calendar/blob/d5f0504a24e463d6373ef772f244462a2af35dfd/src/Tribe/Front_Page_View.php#L162)
:warning: | The method parameter $name is never used | [Tribe__Events__Customizer__Front_Page_View.php:21](https://github.com/moderntribe/the-events-calendar/blob/d5f0504a24e463d6373ef772f244462a2af35dfd/src/deprecated/Tribe__Events__Customizer__Front_Page_View.php#L21)
:warning: | The method parameter $pages is never used | [Tribe__Events__Customizer__Front_Page_View.php:44](https://github.com/moderntribe/the-events-calendar/blob/d5f0504a24e463d6373ef772f244462a2af35dfd/src/deprecated/Tribe__Events__Customizer__Front_Page_View.php#L44)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $option is never used | [Front_Page_View.php:142](https://github.com/moderntribe/the-events-calendar/blob/4e3172a507dcf2981c722b82a0e665adf890e219/src/Tribe/Front_Page_View.php#L142)
:warning: | The method parameter $pages is never used | [Front_Page_View.php:162](https://github.com/moderntribe/the-events-calendar/blob/4e3172a507dcf2981c722b82a0e665adf890e219/src/Tribe/Front_Page_View.php#L162)
:warning: | The method parameter $name is never used | [Tribe__Events__Customizer__Front_Page_View.php:21](https://github.com/moderntribe/the-events-calendar/blob/4e3172a507dcf2981c722b82a0e665adf890e219/src/deprecated/Tribe__Events__Customizer__Front_Page_View.php#L21)
:warning: | The method parameter $pages is never used | [Tribe__Events__Customizer__Front_Page_View.php:44](https://github.com/moderntribe/the-events-calendar/blob/4e3172a507dcf2981c722b82a0e665adf890e219/src/deprecated/Tribe__Events__Customizer__Front_Page_View.php#L44)

> via jenkins/codesniffer-phpcs

**Pull request merged**